<launch>
  
  <node name="odom_pub" pkg="tutorial" type="odom_pub.py" >
  </node>
  <!--node name="pwm_pub" pkg="tutorial" type="ros_pwm.py" >
  </node-->
  <node name="imu_node" pkg="tutorial" type="imu_node.py" >
  </node>
  <node name="imu_odom" pkg="tutorial" type="imu_odom.py" >
  </node>

  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
  <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
  <param name="serial_baudrate"     type="int"    value="115200"/>
  <param name="frame_id"            type="string" value="laser"/>
  <param name="inverted"            type="bool"   value="false"/>
  <param name="reversion"    type="bool"   value="true"/>
  <param name="angle_min"    type="double"   value="-180"/>
  <param name="angle_max"    type="double"   value="180"/>
  <param name="range_min"    type="double"   value="0.01"/>
  <param name="range_max"    type="double"   value="32.0"/>
  <param name="frequency"    type="double"   value="10"/>
  <param name="samp_rate"    type="int"   value="20"/>
  </node>
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_base_footprint" args="-0.0 0.0 0.0 0.0 0.0 0.0 base_footprint base_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser" args="-0.1 0.0 0.2 0.0 0.0 0.0 base_link laser 100"/>
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_imu" args="-0.1 0.1 0.2 0.0 0.0 0.0 base_link imu_link 100"/>
  

</launch>
